### ISSUE TITLE
Url does not pass as expected

### ISSUE THREAD

**Issue by digitalanalogue** [patternhelloworld/url-knife#8]
This url: https://www.php.net/manual/en/language.oop5.overloading.php#object.call

Gives these results:

`{
  "value": {
    "url": "https://www.php.net/manual/en/language.oop5.overloading.php#object.call",
    "normalizedUrl": "https://www.ph/manual/en/language.oop5.overloading.php#object.call",
    "removedTailOnUrl": "",
    "protocol": "https",
    "onlyDomain": "www.ph",
    "onlyParams": null,
    "onlyUri": "/manual/en/language.oop5.overloading.php#object.call",
    "onlyUriWithParams": "/manual/en/language.oop5.overloading.php#object.call",
    "onlyParamsJsn": null,
    "type": "domain",
    "port": null
  },
  "area": "text"
}`

When used like so:

`Pattern.TextArea.extractAllFuzzyUrls('https://www.php.net/manual/en/language.oop5.overloading.php#object.call');` 

I'm using this within a browser (firefox 86 on OSX).

I expect the onlyDomain to be www.php.net 

Great project btw, just found this bug when messing about with pasting in urls.

**Comment by digitalanalogue** [comment]
Also found domains with hypens like this http://sub.do-main.co.uk/ do not work and come out like http://sub.do.main.co.uk/

**Comment by Andrew-Kang-G** [comment]
You are right. **extractAllUrls is currently a recommended method to extract URLs**, but extractAllFuzzyUrls is kind of an experimental thing to extract corrupted ones. I will look into extractAllFuzzyUrls for the issue later.
